### ISSUE TITLE
.Renviron not being processed properly

### ISSUE THREAD

**Issue by dmurdoch** [rstudio/rstudio#9458]
### System details

    RStudio Edition : Desktop
    RStudio Version :  1.4.1103
    OS Version      : macOS 10.15.7 Catalina
    R Version       :  4.0.5 patched (with debugging support)

### Steps to reproduce the problem

Create a new R Package project.  In the R subdir, add a file `msg.R` containing this one line:

    message("TESTENVVAR=", Sys.getenv("TESTENVVAR", "unset"))

In the root dir of the package, add a file `.Renviron` containing the line

    TESTENVVAR=

In your home directory, add a file `.Renviron` containing the line

    TESTENVVAR=home

Close RStudio and re-open to this new project, and install it.  Somewhere in the output you'll see

    ** byte-compile and prepare package for lazy loading
    TESTENVVAR=home
    No man pages found in package  ‘testpkg’ 

If you go into the terminal, set wd to the package directory, and run `R CMD INSTALL .`, you'll see

```
** byte-compile and prepare package for lazy loading
TESTENVVAR=unset
** help
```

as you should.

If you change the `.Renviron` that's in the package to say

    TESTENVVAR=local

you'll see

```
** byte-compile and prepare package for lazy loading
TESTENVVAR=local
No man pages found in package  ‘testpkg’ 
```

as you should in both cases.

### Describe the problem in detail

When the `.Renviron` file tries to unset an environment variable, it is ignored.  

### Describe the behavior you expected

The `.Renviron` file settings of a blank value should either set the variable to `""` or unset it.

- [x ] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x ] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x ] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [ x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by ronblum** [comment]
@dmurdoch Thank you for raising this. The examples you provide seem to describe what you're expecting. What does the result that you're not expecting look like?

**Comment by dmurdoch** [comment]
I was not expecting to see `TESTENVVAR=home` in the install log when the local `.Renviron` file contained `TESTENVVAR=`.

**Comment by kevinushey** [comment]
My suspicion is that there is a conflict in the environment variables being saved and restored by RStudio, and the environment variables being set in the `.Renviron`. When Build and Install is executed, the flow is something like this:

1. Suspend the session (saving session state; e.g. environment variables)
2. Restart the R session (thereby asking R to run `.Renviron`),
3. Restore the session state (restoring e.g. environment variables)

The problem here is that the environment variables from the previous session state (that is, when TESTENVVAR=home) is overwriting the new value set by .Renviron.

As a workaround, explicitly restarting the R session should tell RStudio to discard the saved environment variables and read whatever is new + current in the `.Renviron`.

**Comment by dmurdoch** [comment]
That workaround doesn't seem to make a difference to me.  What it looks like is that RStudio is processing the .Renviron files itself.  It sees that nothing is set in the local one (when really I want to unset something), and goes to the home dir one to process.

The workaround I've adopted is to avoid trying to unset a variable.  If I set any value in the local one it takes precedence.  

The context was the rgl package, which looked for `nchar(...) > 0` in `RGL_DEBUGGING` to turn on debugging when anything was set there; now I test for `isTRUE(as.logical( ... ))` and can explicitly set `FALSE` to leave debugging turned off.
